### PR TITLE
forc: Add util `user_forc_directory` function

### LINKED ISSUE #778: Compute Forc home directory in only one place
#674 uses manually computed home directory paths. The Forc home directory (default `$HOME/.forc/`) should be a thing that's computed/set once, and usable everywhere in the codebase.

### PR DESCRIPTION
**Pull request by mitchmindtree**
This changes the `forc explorer` tool from using `.fuel` as its user
directory to follow suit of `forc`'s dependency handling and use `.forc`
instead.

Path handling for both has also been fixed to theoretically better
support windows (though I haven't tested).

Both are now implemented in terms of the new `user_forc_directory`
function.

Closes #778.

### CODE REVIEW

File: forc/src/utils/helpers.rs [hunk 1]
```diff
@@ -76,6 +76,13 @@ pub fn default_output_directory(manifest_dir: &Path) -> PathBuf {
     manifest_dir.join(DEFAULT_OUTPUT_DIRECTORY)
 }
 
+/// Returns the user's `.forc` directory, `$HOME/.forc` by default.
+pub fn user_forc_directory() -> PathBuf {
+    dirs::home_dir()
+        .expect("unable to find the user home directory")
+        .join(".forc")
```
Comment (adlerjohn) [inline forc/src/utils/helpers.rs:83]: Should use `FORC_DEPENDENCIES_DIRECTORY` here. Maybe rename the constant to something more general.

https://github.com/FuelLabs/sway/blob/be7b2d48b0afe6427bf81482e663ae17773fe6d9/sway-utils/src/constants.rs#L5

Comment (mitchmindtree) [reply forc/src/utils/helpers.rs]: Nice, renaming this helped to find a few more spots that could be refactored in terms of the new function too :+1:

**Review by adlerjohn — APPROVED** [review]
Haven't tested, but seems fine to me.

**Review by alex-matthe — APPROVED** [review]
Looks good to me! The explorer paths struct looks cleaner now

**Pull request merged**